### PR TITLE
Added static logger to remove actions core logging dependency

### PR DESCRIPTION
**Pull request by anricha**

### CODE REVIEW

**Comment by lordozb** [comment]
This is great
Can we also add unit tests?

**Comment by anricha** [comment]
> t

What did you have in mind for the tests?

**Comment by anricha** [comment]
> This is great
> Can we also add unit tests?

Added unit tests for SystemLogger

File: build_and_deploy/main.ts [hunk 1]
```diff
@@ -16,31 +17,30 @@ export async function main() {
...
-
 }
 
 main()
     .then(() => {
         process.exit(0)
     })
     .catch((err: Error) => {
-        core.info("Action failed -> "+ err);
+        core.info("Action failed -> " + err);
```
Comment (praveenmathamsetty) [inline build_and_deploy/main.ts:43]: can't we use system logger here?

File: build_and_deploy/main.ts [hunk 2]
```diff
@@ -2,11 +2,12 @@
 // Licensed under the MIT license.
 
 
+import * as core from '@actions/core';
```
Comment (praveenmathamsetty) [inline build_and_deploy/main.ts:5]: Is it possible to avoid actions core?

File: build_and_deploy/clients/artifacts_client.ts [hunk 3]
```diff
@@ -2,12 +2,13 @@
 // Licensed under the MIT license.
 
 
-import {Params, DeployStatus, Env, getParams} from '../utils/deploy_utils';
+import * as core from '@actions/core';
```
Comment (praveenmathamsetty) [inline build_and_deploy/clients/artifacts_client.ts:5]: we can remove this?

**Pull request merged**